src/support.rs

```diff
@@ -0,0 +1,87 @@
+use crate::palloc::{Inner, InternalAsValue, ToInternal};
+use pgx::*;
+
+// FIXME: This rough translation from C to Rust is _untested_ and probably broken
+
+#[pg_extern(immutable, strict)]
+#[search_path(@extschema@)]
+pub unsafe fn make_call_subquery_support(input: Internal) -> Internal {
+    make_call_subquery_support_inner(input.to_inner()).internal()
+}
+
+pub unsafe fn make_call_subquery_support_inner(
+    input: Option<Inner<*mut pg_sys::Node>>,
+) -> Inner<*mut pg_sys::Node> {
```

Comment:
I _think_ this should be
```suggestion
    input: Option<Inner<pg_sys::Node>>,
) -> Internal {
```
`Inner` is already a pointer, and is guaranteed non-null, so it's not a valid return type for this.
(Honestly, `Inner<...>` is very overfit to aggregate transition states, I suspect just working with `Internal` for both input and output is right)

Reply:
I have reworked this to just use `Internal` as you suggested.